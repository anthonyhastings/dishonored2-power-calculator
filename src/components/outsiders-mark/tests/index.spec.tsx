import renderer from 'react-test-renderer';
import OutsidersMark from '../';

describe('OutsidersMark component', () => {
  let testContext: {
    component?: renderer.ReactTestRenderer;
  };

  const renderComponent = (props = {}) => <OutsidersMark {...props} />;

  beforeEach(() => {
    testContext = {};
  });

  describe('when rendered', () => {
    beforeEach(() => {
      testContext.component = renderer.create(renderComponent());
    });

    it('renders with no modifier class or provided class', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe('when animated is true', () => {
    beforeEach(() => {
      testContext.component = renderer.create(
        renderComponent({
          animated: true,
        })
      );
    });

    it('renders with modifier class', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe('when className is provided', () => {
    beforeEach(() => {
      testContext.component = renderer.create(
        renderComponent({
          className: 'hello-world',
        })
      );
    });

    it('renders with extra className', () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });
});
