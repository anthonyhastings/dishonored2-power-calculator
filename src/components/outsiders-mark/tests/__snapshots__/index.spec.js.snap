// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutsidersMark component when animated is true renders with modifier class 1`] = `
<div
  aria-hidden="true"
  className="outsiders-mark outsiders-mark--is-animating"
/>
`;

exports[`OutsidersMark component when className is provided renders with extra className 1`] = `
<div
  aria-hidden="true"
  className="outsiders-mark hello-world"
/>
`;

exports[`OutsidersMark component when rendered renders with no modifier class or provided class 1`] = `
<div
  aria-hidden="true"
  className="outsiders-mark"
/>
`;
